Resolve mypy errors on default branch

maxsvetlik opened the pull request:
Primarily, this tests branch protections (see [8d12297](https://github.com/maxsvetlik/balatro-gym/pull/20/commits/8d12297e84d1ec72f4e7b97bd9baf159d7a5796a) in which GH correctly prevented merging due to failed tests).

Otherwise:
- Fixes mypy error on `main`
- Gives more intuitive names to the GHA jobs, which is what is shown on the UI

Closes #7

andreanne-lemay reviewed (approved):
lgtm! I fixed the issue on my branch too, but I still can't merge the branch, because I have 6 pending checks, do you understand why merging is being blocked: https://github.com/maxsvetlik/balatro-gym/pull/19?

maxsvetlik commented:
> lgtm! I fixed the issue on my branch too, but I still can't merge the branch, because I have 6 pending checks, do you understand why merging is being blocked: #19?

I updated the branch protection rules so that this PR would pass and could be merged.. but then I dragged my heals. Once this PR merge, #19 should be mergable after merging in the changes from `main`. Sorry about that
